File: .github/scripts/run_claude.py
#!/usr/bin/env python3
"""Entry point for running Claude agent from GitHub Actions."""

import asyncio
import json
import os
import sys

# Add src to path for imports
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "..", "src"))

from claude_runner import run_claude_chunked


async def main():
    issue_title = os.environ.get("ISSUE_TITLE")
    issue_body = os.environ.get("ISSUE_BODY")
    api_key = os.environ.get("ANTHROPIC_API_KEY")

    if not api_key:
        print("Error: ANTHROPIC_API_KEY environment variable is required", file=sys.stderr)
        sys.exit(1)

    if not issue_title:
        print("Error: ISSUE_TITLE environment variable is required", file=sys.stderr)
        sys.exit(1)

    # Allow empty body if title is provided
    if issue_body is None:
        issue_body = ""

    try:
        cwd = os.getcwd()

        # Run in chunks of 10 turns, up to 5 chunks (50 turns max)
        async for message in run_claude_chunked(issue_title, issue_body, cwd):
            print(json.dumps(message, default=str))

    except Exception as e:
        print(f"Error running Claude: {e}", file=sys.stderr)
        sys.exit(1)


if __name__ == "__main__":
    asyncio.run(main())

File: .github/scripts/run-claude.mjs
#!/usr/bin/env node

import { buildPrompt, runClaude } from "../../src/claude-runner.mjs";

const issueTitle = process.env.ISSUE_TITLE;
const issueBody = process.env.ISSUE_BODY;
const apiKey = process.env.ANTHROPIC_API_KEY;

if (!apiKey) {
  console.error("Error: ANTHROPIC_API_KEY environment variable is required");
  process.exit(1);
}

if (!issueTitle || !issueBody) {
  console.error("Error: ISSUE_TITLE and ISSUE_BODY environment variables are required");
  process.exit(1);
}

try {
  const prompt = buildPrompt(issueTitle, issueBody);
  const result = await runClaude(prompt);

  for await (const message of result) {
    console.log(JSON.stringify(message));
  }

  process.exit(0);
} catch (error) {
  console.error("Error running Claude:", error.message);
  process.exit(1);
}
